services/web/server/tests/unit/isolated/conftest.py

```diff
@@ -0,0 +1,65 @@
+from pathlib import Path
+import string
+import secrets
+import random
+import os
+import tempfile
+from typing import Iterator
+
+import pytest
+
+
+@pytest.fixture
+def dir_with_random_content() -> Path:
+    def random_string(length: int) -> str:
+        return "".join(secrets.choice(string.ascii_letters) for i in range(length))
+
+    def make_files_in_dir(dir_path: Path, file_count: int) -> None:
+        for _ in range(file_count):
+            (dir_path / f"{random_string(8)}.bin").write_bytes(
+                os.urandom(random.randint(1, 10))
+            )
+
+    def ensure_dir(path_to_ensure: Path) -> Path:
+        path_to_ensure.mkdir(parents=True, exist_ok=True)
+        return path_to_ensure
+
+    def make_subdirectory_with_content(subdir_name: Path, max_file_count: int) -> None:
+        subdir_name = ensure_dir(subdir_name)
+        make_files_in_dir(
+            dir_path=subdir_name,
+            file_count=random.randint(1, max_file_count),
+        )
+
+    def make_subdirectories_with_content(
+        subdir_name: Path, max_subdirectories_count: int, max_file_count: int
+    ) -> None:
+        subdirectories_count = random.randint(1, max_subdirectories_count)
+        for _ in range(subdirectories_count):
+            make_subdirectory_with_content(
+                subdir_name=subdir_name / f"{random_string(4)}",
+                max_file_count=max_file_count,
+            )
+
+    def get_dirs_and_subdris_in_path(path_to_scan: Path) -> Iterator[Path]:
+        return [path for path in path_to_scan.rglob("*") if path.is_dir()]
+
+    with tempfile.TemporaryDirectory() as temp_dir:
```

Comment:
As we discussed, for testing is more practical using ``tmpdir`` fixture provided by pytest. It creates a tmp under folders with very readable names that are very handy for debugging.  It is as simple as

```python
@fixture
def dir_with_random_content(tmpdir):
# ...
  # no more with tempfile.Temporary....
   data_container = ensure_dir(temp_dir_path / "study_data")
   # ...
   return temp_dir_path
 ```